src/compiler/objective_c_generator.cc

```diff
@@ -176,65 +176,71 @@ void PrintMethodImplementations(Printer *printer,
 
 string GetHeader(const ServiceDescriptor *service, const string prefix) {
   string output;
-  grpc::protobuf::io::StringOutputStream output_stream(&output);
-  Printer printer(&output_stream, '$');
+  {
+    // Scope the output stream so it closes and finalizes output to the string.
```

Comment:
Yes, so. If you do:

```
A a;
B * b = new B(&a);
```
then the constructor of B is taking the address of a temporary. That is supposed to be a warning. I guess it doesn't happen here because we're doing everything on the stack. But still.

For the destruction order, yes, it is determined. The problem here is that the destruction occurs AFTER the move operator is being called to move the variable `output`. While technically a move operation isn't a destruction, it is supposed to leave the object in a destructed state. Meaning that the actual destruction of `output` happens before the actual destruction of `output_stream` and `printer`, which is against the spec.